src/main/com/yetanalytics/datasim/xapi/statement.clj

```diff
@@ -134,7 +134,14 @@
     pattern-ancestors :pattern-ancestors
     registration :registration
     ?sub-registration :sub-registration}]
-  (let [rng        (random/seed-rng seed)
+  (let [rng          (random/seed-rng seed)
+        obj-override (when (not-empty alignment)
+                       (->> alignment
+                            keys
+                            (random/choose rng alignment)
+                            (get alignment)
+                            :object-override
+                            w/stringify-keys))
```

Comment:
I'm not sure about just the conditional here, the thread macro is a little dangerous, even if it doesn't usually throw. I'd suggest using `clojure.core/some->>`

Also `obj-override` is nullable here right? I generally like to indicate that with the convention `?obj-override` but opinions vary on that

Reply:
you could do the right side of the binding as:

    (some->> alignment
                      not-empty ...)

Reply:
Yes thanks for reminding me about the `?` convention. And I guess I could change `->>` to `some->>` (though I don't think it should throw thanks to the `not-empty` guard).

Reply:
yeah, I mean it just so happens that `w/stringify-keys` is cool with nils, but not everything is so forgiving. I was talking more about the fact that :object-override is optional on alignments